src/latbin/lattice-lmrescore-kaldi-rnnlm.cc

```diff
@@ -0,0 +1,154 @@
+// latbin/lattice-lmrescore-kaldi-rnnlm.cc
+
+// Copyright 2017 Johns Hopkins University (author: Daniel Povey)
+//           2017 Hainan Xu
+//           2017 Yiming Wang
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "base/kaldi-common.h"
+#include "fstext/fstext-lib.h"
+#include "lat/kaldi-lattice.h"
+#include "lat/lattice-functions.h"
+#include "rnnlm/rnnlm-lattice-rescoring.h"
+#include "util/common-utils.h"
+#include "nnet3/nnet-utils.h"
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+    typedef kaldi::int32 int32;
+    typedef kaldi::int64 int64;
+
+    const char *usage =
+        "Rescores lattice with kaldi-rnnlm. This script is called from \n"
+        "scripts/rnnlm/lmrescore_rnnlm_lat.sh. An example for rescoring \n"
+        "lattices is at egs/swbd/s5/local/rnnlm/run_rescoring.sh \n"
+        "\n"
+        "Usage: lattice-lmrescore-kaldi-rnnlm [options] \\\n"
+        "             <embedding-file> <raw-rnnlm-rxfilename> \\\n"
+        "             <lattice-rspecifier> <lattice-wspecifier>\n"
+        " e.g.: lattice-lmrescore-kaldi-rnnlm --lm-scale=-1.0 \\\n"
+        "              word_embedding.mat \\\n"
+        "              --bos-symbol=1 --eos-symbol=2 \\\n"
+        "              final.raw ark:in.lats ark:out.lats\n";
+
+    ParseOptions po(usage);
+    nnet3::RnnlmComputeStateComputationOptions opts;
+
+    int32 max_ngram_order = 3;
+    BaseFloat lm_scale = 1.0;
+
+    po.Register("lm-scale", &lm_scale, "Scaling factor for language model "
+                "costs");
+    po.Register("max-ngram-order", &max_ngram_order,
+        "If positive, allow RNNLM histories longer than this to be identified "
+        "with each other for rescoring purposes (an approximation that "
+        "saves time and reduces output lattice size).");
+    opts.Register(&po);
+
+    po.Read(argc, argv);
+
+    if (po.NumArgs() != 4) {
+      po.PrintUsage();
+      exit(1);
+    }
+
+    if (opts.bos_index == -1 || opts.eos_index == -1) {
+      KALDI_ERR << "must set --bos-symbol and --eos-symbol options";
```

Comment:
error messages should be properly capitalized and be proper sentences.  Change to "You must set the " at the beginning.